src/Quaternion.jl

```diff
@@ -207,9 +207,14 @@ function qrotation(axis::Vector{T}, theta) where {T <: Real}
     if length(axis) != 3
         error("Must be a 3-vector")
     end
-    u = normalize(axis)
-    s = sin(theta / 2)
-    Quaternion(cos(theta / 2), s * u[1], s * u[2], s * u[3], true)
+    normaxis = norm(axis)
+    if iszero(normaxis)
+        normaxis = one(normaxis)
```

Comment:
This ensures type-stability if the eltype of `axis` is a dimensional number
```suggestion
        normaxis = oneunit(normaxis)
```